neuro_san/service/grpc/grpc_agent_service.py

```diff
@@ -60,18 +61,23 @@ def __init__(self,
                         their logging.
         """
         self.forwarder: GrpcMetadataForwarder = server_logging.get_forwarder()
-        self.service: AgentService =\
-            AgentService(request_logger,
-                         security_cfg,
-                         agent_name,
-                         agent_network_provider,
-                         server_logging)
+        self.service_provider: AgentServiceProvider =\
+            AgentServiceProvider(
+                request_logger,
+                security_cfg,
+                agent_name,
+                agent_network_provider,
+                server_logging)
 
     def get_request_count(self) -> int:
         """
         :return: The number of currently active requests
         """
-        return self.service.get_request_count()
+        if not self.service_provider.service_created():
+            # Service is not yet instantiated - it has no requests
+            return 0
+        service: AgentService = self.service_provider.get_service()
+        return service.get_request_count()
```

Comment:
That's perf optimization: if service instance has not been instantiated, then its request count is obviously 0.
Otherwise with current gRPC server logic, we'll always instantiate all agent services,
even when not using gRPC service at all.